pkg/exthttp/hedging.go

```diff
@@ -0,0 +1,73 @@
+package exthttp
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/caio/go-tdigest"
+	"github.com/cristalhq/hedgedhttp"
+)
+
+type CustomBucketConfig struct {
+	HedgingConfig HedgingConfig `yaml:"hedging_config"`
+}
+
+type HedgingConfig struct {
+	UpTo uint `yaml:"up_to"`
+}
+
+var HedgedOptions CustomBucketConfig
+
+type durationRoundTripper struct {
+	Transport http.RoundTripper
+	TDigest   *tdigest.TDigest
+	mu        sync.Mutex
```

Comment:
I feel like we'll see some lock contention here, across different goroutines, especially as we hold to add and calculate quantile.
I wonder if we can somehow send these data points through a channel and let it consume those?

Or maybe simpler thing would an RWMutex?

Reply:
Yes, maybe then lets try RWMutex